[CI] TSDBSyntheticIdUpgradeIT testRollingUpgrade {upgradedNodes=2} failing

**Build Scans:**
- [elasticsearch-intake #34092 / bc-upgrade-tests-part6](https://gradle-enterprise.elastic.co/s/bpo5rdlqceh3o)
- [elasticsearch-pull-request #118229 / 9.3.0_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/os633k42bnffa)
- [elasticsearch-pull-request #117624 / 9.1.11_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/5y56egg3lfrky)
- [elasticsearch-pull-request #117406 / 9.3.0_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/4egnea5m3iotu)
- [elasticsearch-pull-request #117406 / 9.1.11_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/s6txyezfaekhk)

**Reproduction Line:**
```
./gradlew ":qa:rolling-upgrade:bcUpgradeTest" -Dtests.class="org.elasticsearch.upgrades.TSDBSyntheticIdUpgradeIT" -Dtests.method="testRollingUpgrade {upgradedNodes=2}" -Dtests.seed=A7F8AEF8482BE5FF -Dtests.bwc.main.version=9.4.0-SNAPSHOT -Dtests.bwc.refspec.main=e2d675ecda7274c4f368e3c43491ceb170095c5d -Dtests.locale=pa-Guru -Dtests.timezone=America/Phoenix -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.upgrades.TSDBSyntheticIdUpgradeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRollingUpgrade%20%7BupgradedNodes%3D2%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: (a string containing "unknown setting [index.mapping.synthetic_id]" or a string containing "The setting [index.mapping.synthetic_id] is set to [true] but index metadata has a different value [false]")
     but: was "method [PUT], host [http://[::1]:39865], URI [/mixed-cluster-index], status line [HTTP/1.1 400 Bad Request]\n---\nerror:\n  root_cause:\n  - type: \"illegal_argument_exception\"\n    reason: \"The setting [index.mapping.synthetic_id] is set to [true] but index metadata\\\n      \\ has a different value [false].\"\n  type: \"illegal_argument_exception\"\n  reason: \"The setting [index.mapping.synthetic_id] is set to [true] but index metadata\\\n    \\ has a different value [false].\"\nstatus: 400\n"
```

**Issue Reasons:**
- [main] 5 failures in test testRollingUpgrade {upgradedNodes=2} (9.3% fail rate in 54 executions)
- [main] 2 failures in step 9.3.0_part_6_bwc-snapshots (22.2% fail rate in 9 executions)
- [main] 2 failures in step 9.1.11_part_6_bwc-snapshots (22.2% fail rate in 9 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (33.3% fail rate in 9 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -437,12 +437,6 @@ tests:
 - class: org.elasticsearch.index.reindex.ReindexResumeIT
   method: testLocalResumeReindexFromScroll_slicedAuto
   issue: https://github.com/elastic/elasticsearch/issues/142749
-- class: org.elasticsearch.upgrades.TSDBSyntheticIdUpgradeIT
-  method: testRollingUpgrade {upgradedNodes=2}
-  issue: https://github.com/elastic/elasticsearch/issues/141734
-- class: org.elasticsearch.upgrades.TSDBSyntheticIdUpgradeIT
-  method: testRollingUpgrade {upgradedNodes=3}
-  issue: https://github.com/elastic/elasticsearch/issues/141749
 - class: org.elasticsearch.xpack.esql.analysis.AnalyzerTests
   method: testResolveExternalRelationPassesFileSet
   issue: https://github.com/elastic/elasticsearch/issues/142795

--- a/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/TSDBSyntheticIdUpgradeIT.java
+++ b/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/TSDBSyntheticIdUpgradeIT.java
@@ -140,7 +140,7 @@ public class TSDBSyntheticIdUpgradeIT extends AbstractRollingUpgradeTestCase {
             """, timestamp, randomByte()));
     }
 
-    private static void assertNoWriteIndex(String indexName, IndexVersion oldClusterIndexVersion) {
+    private static void assertNoWriteIndex(String indexName, IndexVersion oldClusterIndexVersion) throws IOException {
         String setting = IndexSettings.SYNTHETIC_ID.getKey();
         String unknownSetting = "unknown setting [" + setting + "]";
         String versionTooLow = String.format(
@@ -152,7 +152,9 @@ public class TSDBSyntheticIdUpgradeIT extends AbstractRollingUpgradeTestCase {
         );
 
         ResponseException e = assertThrows(ResponseException.class, () -> createSyntheticIdIndex(indexName));
-        assertThat(e.getMessage(), Matchers.either(Matchers.containsString(unknownSetting)).or(Matchers.containsString(versionTooLow)));
+        String reason = ObjectPath.createFromResponse(e.getResponse()).evaluate("error.reason");
+
+        assertThat(reason, Matchers.either(Matchers.containsString(unknownSetting)).or(Matchers.containsString(versionTooLow)));
         assertThat(e.getMessage(), Matchers.containsString("illegal_argument_exception"));
     }
 

